libethcore/Common.h

```diff
@@ -90,6 +94,81 @@ enum class RelativeBlock: BlockNumber
 	Pending = PendingBlock
 };
 
+enum class BlockPolarity
+{
+	Unknown,
+	Dead,
+	Live
+};
+
+struct LogEntry
+{
+	LogEntry() = default;
+	explicit LogEntry(RLP const& _r);
+	LogEntry(Address const& _address, h256s _ts, bytes&& _d): address(_address), topics(std::move(_ts)), data(std::move(_d)) {}
```

Comment:
I wonder why `_ts` is not an rvalue reference and `_d` is an rvalue reference...

Reply:
Generic way is to pass by value and move in the initialization list (like with `h256s` case). This way you handle both copy and move, depending what argument you have.

The version with `&&` prevents you from passing argument by copy -- you can only pass an argument being rvalue. This is restrictive, but enforces some optimizations. I don't like it because it is more confusing than useful.